src/nvim/path.c

```diff
@@ -2196,11 +2202,11 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
   // expand it if forced or not an absolute path
   if (force || !path_is_absolute_path(fname)) {
-    if ((p = vim_strrchr(fname, '/')) != NULL) {
+    if ((p = vim_strrchr(fname, PATHSEP)) != NULL) {
```

Comment:
It looks like `slash_adjust()` isn't always guaranteed to have already been called by now, so there could be a `/` slash here on Windows. It  probably makes sense to check for both here. Maybe we need a new function in the style of `slash_adjust` :

(untested...)

```c
char *slash_index(char *fname)
{
  char *p = NULL;
#ifdef BACKSLASH_IN_FILENAME
  p = vim_strrchr(psepc);
  if (p == NULL) {
    p = vim_strrchr(psepcN);
  }
#else
  p = vim_strrchr(PATHSEP);
#endif
  return p;
}
```

`path_fix_case` has another `vim_strrchr(name, '/')` case.

In these cases

Reply:
Not sure about this though , so maybe different PR.

Reply:
When would that would be the case? Because if we can write a test for it and this is inside the scope of the expand('%:p') case, then I would be willing to do it before the merge if it is a different case, then it is maybe better to open an issue and address it into a separate PR. :)

Reply:
I don't know. Finding a test case for this is the hard part :)